cri-purge.sh

```diff
@@ -78,24 +81,26 @@ __determine_containerd_root_dir() {
 __generate_image_list() {
 
   # load list of images / filter out header line / version sort on 60th char of line
-  CRI_IMAGES=$(${CRI_CMD} images | tail -n +2 | sort -k 1.60 -V)
+  ${CRI_CMD} images | tail -n +2 | sort -k 1.60 -V > /tmp/CRI_IMAGES
 
   # Filter out TAGS to SKIP
   for TAG in ${SKIP_THESE_TAGS};
   do
-    if [ "$(echo "${CRI_IMAGES}" | grep -c "${TAG}")" -ne 0 ]; then
-      echo "NOTE: Skipping Images with Tag: ${TAG}"
-      echo "${CRI_IMAGES}" | grep "${TAG}"
-      CRI_IMAGES=$(echo "${CRI_IMAGES}" | grep -v "${TAG}") 
+    grep -oP "${TAG}" /tmp/CRI_IMAGES > /tmp/CRI_IMAGES_SKIP
```

Comment:
I'll rework this section some more as what you impliment seems opposite of intended. The "grep -vP" generates the list of images that would have been skipped (images with "none" tag) and "grep -oP" generates the list of images with tags.

What I'm going to do is keep the images with "none" tag as being skipped by default. But I will add a new flag to enable processing "none" tags and will then purge based on "hash" as you provided.